server/backend/api/app/crud/crud_user_role.py

```diff
@@ -298,3 +297,37 @@ def find_by_id(db: Session, user_role_xref_id: int) -> models.FamUserRoleXref:
         .one_or_none()
     )
     return user_role
+
+
+def send_user_access_granted_email(
+        target_user: schemas.TargetUser,
+        roles_assigned: List[schemas.FamUserRoleAssignmentCreateResponse]):
+    """
+    Send email using GC Notify integration service.
+    TODO: Erro handling when sending email encountered technical errors (400/500). Ticket #1471.
+        - do not fail event when sending email fails (400/500).
+        - 'create_user_role_assignment_many' router's response schema needs to be refactored.
+        - if email sent -> include in response status/message email is sent.
+        - if email sent failed (400/500) -> include in response status/message email is not sent
+        - frontend needs to display email sent failure message.
+
+    Note
+        - FAM currently is not concerned with checking status from GC Notify (callback) to verify
+            if email is really sent from GC Notify.
+    """
+    granted_roles = ", ".join(item.detail.role.role_name for item in roles_assigned)
+    email_service = GCNotifyEmailService()
+    email_params = schemas.GCNotifyGrantAccessEmailParam(**{
+        "first_name": target_user.first_name,
+        "last_name": target_user.last_name,
+        "application_name": roles_assigned[0].detail.role.application.application_description,
+        "role_list_string": granted_roles,
```

Comment:
Hi Ian, just remind that, how about in the case the status of granting one role is failed, but the rest are successful? should the email only include the successful roles?

Reply:
Oh yes, good catch, I forgot this method return multiple responses. Thanks, I will adjust that to send email for successful roles granted.